samples/powerpoint/preview-apis/insert-slides.yaml

```diff
@@ -0,0 +1,137 @@
+id: powerpoint-insert-slides
+name: Insert slides from other presentation
+description: Inserts slides from another PowerPoint file into the current presentation.
+host: POWERPOINT
+api_set:
+    PowerPointApi: '1.2'
+script:
+    content: |
+        $("#insert-all-slides").click(() => tryCatch(insertAllSlides));
+        $("#insert-after-target-slide").click(() => tryCatch(insertAfterSelectedSlide));
+        $("#file").change(() => tryCatch(storeFileAsBase64));
+
+        let chosenFileBase64;
+
+        async function storeFileAsBase64() {
+            const reader = new FileReader();
+
+            reader.onload = async (event) => {
+                // Strip off the metadata before the base64-encoded string
+                const startIndex = reader.result.toString().indexOf("base64,");
+                const copyBase64 = reader.result.toString().substr(startIndex + 7);
+
+                chosenFileBase64 = copyBase64;
+            };
+
+            // Read in the file and store a base64-encoded copy as the reader.result
+            // property. This also triggers the onload event.
+            const myFile = document.getElementById("file") as HTMLInputElement;
+            reader.readAsDataURL(myFile.files[0]);
+        }
+
+        async function insertAllSlides() {
+            await PowerPoint.run(async function(context) {
+                context.presentation.insertSlidesFromBase64(chosenFileBase64);
+                await context.sync();
+            });
+        }
+
+        async function insertAfterSelectedSlide() {
+            await PowerPoint.run(async function(context) {
+                const selectedSlideID = await getSelectedSlideID();
+
+                context.presentation.insertSlidesFromBase64(chosenFileBase64, {
+                    formatting: PowerPoint.InsertSlideFormatting.useDestinationTheme,
+                    targetSlideId: selectedSlideID + "#"
+                });
+                await context.sync();
+            });
+        }
+
+        function getSelectedSlideID() {
+            // Wrap a call of one of the Common APIs in a Promise-returning
+            // function, so that it can be easily called within a run() method
+            // of an application-specific API.
+            return new OfficeExtension.Promise<string>(function(resolve, reject) {
+                Office.context.document.getSelectedDataAsync(Office.CoercionType.SlideRange, function(asyncResult) {
+                    try {
+                        if (asyncResult.status === Office.AsyncResultStatus.Failed) {
+                            reject(console.error(asyncResult.error.message));
+                        } else {
+                            resolve(asyncResult.value.slides[0].id);
```

Comment:
This line doesn't compile in Script Lab. Consider casting `asyncResult.value` to remove the "Property 'slides' does not exist on type 'unknown'." message.